zerver/lib/bugdown/__init__.py

```diff
@@ -1268,12 +1270,19 @@ def make_realm_emoji(src: str, display_string: str) -> Element:
     return elt
 
 def unicode_emoji_to_codepoint(unicode_emoji: str) -> str:
-    codepoint = hex(ord(unicode_emoji))[2:]
-    # Unicode codepoints are minimum of length 4, padded
-    # with zeroes if the length is less than zero.
-    while len(codepoint) < 4:
-        codepoint = '0' + codepoint
-    return codepoint
+    def process_char(codepoint_str: str) -> str:
+        codepoint = hex(ord(codepoint_str))[2:]
+        # Unicode codepoints are minimum of length 4, padded
+        # with zeroes if the length is less than zero.
+        while len(codepoint) < 4:
+            codepoint = '0' + codepoint
+        return codepoint
```

Comment:
FYI this whole loop (which I see is in the existing code here too) can be simplified to:
`return '%04x' % (ord(codepoint_str),)`

For example:
```
>>> '%04x' % ord(' ')
'0020'
>>> '%04x' % ord('🤔')
'1f914'
```